receiver/awsnvmereceiver/scraper.go

```diff
@@ -17,35 +16,31 @@ import (
 	"go.uber.org/zap"
 
 	"github.com/aws/amazon-cloudwatch-agent/internal/util/collections"
+	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/ebs"
+	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/instancestore"
 	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/metadata"
 	"github.com/aws/amazon-cloudwatch-agent/receiver/awsnvmereceiver/internal/nvme"
 )
 
-const (
-	ebsModel           = "Amazon Elastic Block Store"
-	instanceStoreModel = "Amazon EC2 NVMe Instance Storage"
-)
-
 type nvmeScraper struct {
 	logger *zap.Logger
 	mb     *metadata.MetricsBuilder
 	nvme   nvme.DeviceInfoProvider
 
-	allowedDevices       collections.Set[string]
-	collectEbs           bool
-	collectInstanceStore bool
+	allowedDevices  collections.Set[string]
+	enabledScrapers []nvme.DeviceTypeScraper
```

Comment:
Why do we need to store the slice? It's never used outside of populating the `scrapersByModel`. Would suggest renaming `scrapersByModel` to `enabledScrapers` and removing this field.